src/main/scala/fi/oph/kouta/service/koulutusService.scala

```diff
@@ -109,7 +109,7 @@ class KoulutusService(sqsInTransactionService: SqsInTransactionService, val s3Se
     KoutaDatabase.runBlockingTransactionally {
       for {
         _          <- KoulutusDAO.checkNotModified(koulutus.oid.get, notModifiedSince)
-        (teema, k) <- checkAndMaybeCopyTempImage(koulutus)
+        (teema, k) <- checkAndMaybeCopyTeemakuva(koulutus)
         k          <- KoulutusDAO.getUpdateActions(k)
         _          <- maybeDeleteTempImage(teema)
         _          <- index(k)
```

Comment:
Ihan vaan yleisesti jäin pohtimaan koskien ihan kaikkia näitä transaktioita, että jos esim. indeksointi tai audit-lokitus jostain syystä feilaa ja aiheuttaa rollbackin, niin siinä tapauksessa kanta vs. s3 voinee mennä epäsynkkaan, jos esim. teemakuvia on ehditty deletoimaan eikä sitä operaatioita oikeasti rollbackata. Ei sille ehkä nyt tarvinne tehdä mitään, mutta sitä voinee pohtia, että tuleeko siitä ongelmia ja pitäisikö sitä yrittää ratkaista jotenkin.

Reply:
Tuon deleten voisi varmaan siirtää transaktion ulkopuolelle ja tehdä sen vain, jos transaktio onnistuu. Se ei sinänsä riko mitään, jos se jää tekemättä, niin epäonnistumisen ei tarvisi aiheuttaa rollbackia. Temp-kansioon jää vain ylimääräinen tiedosto. Siihen temp-kansioon pitäisi muutenkin laittaa joku siivousrutiini siivoamaan kuvia, jotka on uploadattu, mutta joita ei ole sittemmin käytetty.

Jos transaktio epäonnistuu sen jälkeen, kun kuva on jo kopioitu, ei sekään paljoa haittaa. Kopioitu tiedosto ylikirjoitetaan, jos samalle objektille tallennetaan toinen teemakuva.

Jos audit-logitus epäonnistuu, lähetetään turha indeksointipyyntö, mutta sekään ei riko mitään.